lib/validate.js

```diff
@@ -3,16 +3,24 @@
 const BbPromise = require('bluebird');
 const path = require('path');
 const fse = require('fs-extra');
+const glob = require('glob');
 const lib = require('./index');
 const _ = require('lodash');
 
-function getEntryForFunction(serverlessFunction) {
+function getEntryExtension(fileName, servicePath) {
+  let files = glob.sync(`${fileName}.*`, { cwd: servicePath });
+  return path.extname(files[0]);
```

Comment:
We could emit a warning in case more than one file is found here:
```
if (_.size(files) > 1) {
  this.serverless.cli.log(`WARNING: More than one matching handlers found for ${fileName}. Using ${_.first(files)}.`);
}
```

Of course the now global function(s) have to be moved into validate() to have access to the correct `this`.

Also use `_.first(files)` here instead of `files[0]` as the lodash variant will not crash if no file is found.